### ISSUE TITLE
mouse_support prevents scrollwheel from working on Linux

### ISSUE THREAD

**Issue by mixmastamyk** [jonathanslenders/python-prompt-toolkit#224]
Awesome toolkit !!! 

Can't use the scrollwheel on my mouse after setting this on Linux:

    mouse_support=True

Latest pptk on Ubuntu Wily.

**Comment by jonathanslenders** [comment]
Hi @mixmastamyk,
This depends on the terminal. Could you try holding the shift key while scrolling? It could be that the terminal doesn't send scroll events to the application running in the terminal if shift was pressed.

Jonathan

**Comment by mixmastamyk** [comment]
Hmm, nope, shift does allow pasting etc, but scroll-wheel is not working with or without the shift key.  It is mate (gnome2) terminal, btw.

I've disabled the feature for now since the benefits are not huge and the drawbacks significant.  Perhaps it is just the nature of mouse support, if I had a menu or something I might re-enable.

**Comment by abarnert** [comment]
I don't think this has anything to do with which terminal you're using. I see the same thing with gnome-terminal on linux, iTerm2 on Mac, and rxvt on linux talking to X server on Mac--the terminal only seems to get mouse-2 and mouse-3 events; everything else gets eaten by prompt-toolkit.

It would be really nice if I could use mouse clicks to move the cursor in ptpython, while still being able to scroll up and select history for copying in iTerm2. I'm not sure if there's an obvious design to make that work (how do you pass through mouse-1 drags and double-clicks needed for select, while handling mouse-1 clicks?), but if there is, I'd be happy. :)

**Comment by zpincus** [comment]
I can confirm this on a Mac as well, using the default Apple terminal: mouse integration completely prevents the scrollwheel from scrolling the terminal. 

It would be great of prompt_toolkit only eats scrollwheel events in contexts where that is applicable (e.g. scrolling through completions or similar), but I have no idea if this is easy to distinguish or not.

**Comment by jonathanslenders** [comment]
This is normal. Either the terminal can use the mouse to handle text selection or scrolling through the history lines, or the mouse events are sent to the application running in the terminal.

But most terminals will still handle the mouse when the application requests mouse events, if you hold alt, option, control or shift while scrolling or selecting text.

**Comment by zpincus** [comment]
Sadly, both in ptpython and IPython 5 on my mac, using the latest pip version of prompt_toolkit (on OS X 10.11.5, using standard apple terminal):
- With mouse support enabled, the scroll wheel does not appear to scroll through history or do anything at all *except* when a menu is open.
- none of alt, option, control or shift while scrolling send the mouse event back to the terminal.

It's a pity because otherwise the mouse support is super awesome! (It's pretty impressive that any of that can work in a terminal in the first place.)

**Comment by mangecoeur** [comment]
Ran into this issue on macOS 10.12 when using Xonsh  ( https://github.com/xonsh/xonsh/issues/1930#issuecomment-258228533 ). Is a solution possible (apart from turning off mouse supprt in ptk) ?

**Comment by jonathanslenders** [comment]
@mangecoeur: Which terminal are you using? Some terminals handle scrolling if you press shift or alt.

**Comment by excitedleigh** [comment]
If you're using iTerm, you can tell it not to pass scroll events through to prompt_toolkit by unchecking this checkbox in your profile, under the Terminal tab:

![image](https://user-images.githubusercontent.com/473960/82115299-0f460f80-97a1-11ea-80da-f7f6c6bcd849.png)
